File: contest/yukicoder/265/main.py
#UNDONE
n = input()
d = input()
s = raw_input()

print n, d, s

a = [0 for x in xrange(d+1)]

digits = [str(x+1) for x in xrange(9)]

def diff(xdim, times) :
    if(times == 0) :
        return (1, xdim)
    if(xdim == 0) : # x^0を微分しようとした場合0に
        return (0, 0)
    result = diff(xdim-1, times-1)
        return (xdim * result[0], result[1])

class Parse:
    def __init__(self) :
    def parse(self, index=0, d=0, xdim=0, coef=1) :
        if s[index] == "d" and s[index+1] == "{" :
            parse(index+2, d+1, xdim, coef)
        elif s[index] in digits :
            parse(index+1, d, xdim, int(s[index]))
        elif s[index] == "x" :
            parse(index+1, d, xdim+1, coef)
        elif s[index] == "*" :
            parse(index+1, d, xdim, coef)
        elif s[index] == "+" :
            tmp = diff(xdim, d)
                a[tmp[1]] += coef*tmp[0]

File: contest/yukicoder/265/main.ts
import {parser} from "./parser";

declare function require(x: string): any;

var input = require("fs").readFileSync("/dev/stdin", "utf8").split("\n");
var d:number = +input[1];
var s:string = input[2];

let blank = ():Array<number>=>{
	let result = new Array(d+1);
	for(let i=0; i<=d; i++) result[i] = 0;
	return result;
};

console.log(parser.parse(s, (arg)=>{
	let token = arg.token;
	let tmp:Array<number>;
	if(arg.terminal == true){
		let value = arg.value;
		switch(token){
			case "DIGITS":
				tmp = blank();
				tmp[0] = +value;
				return tmp;
			case "X":
				tmp = blank();
				tmp[1] = 1;
				return tmp;
		}
	}
	else{
		let c = arg.children;
		let pattern = arg.pattern;
		switch(token){
			case "EXP":
				if(c.length == 1){
					return c[0];
				}
				else{
					tmp = c[0];
					for(let i=0; i<=d; i++){
						tmp[i] += c[2][i];
					}
				}
			case "DIVTERM":
				return c[0];
			case "TERM":
				if(c.length == 1){
					return c[0];
				}
				else{
					tmp = blank();
					// c[2]はATOMなので[1]==1 or [0]>0のいずれか
					if(c[2][1] == 1){
						for(let i=0; i<d; i++){
							tmp[i+1] = c[0][i];
						}
					}
					else if(c[2][0] > 0){
						for(let i=0; i<=d; i++){
							tmp[i] = c[0][i]*c[2][0];
						}
					}
					return tmp;
				}
			case "ATOM":
				return c[0];
			case "DIV":
				tmp = blank();
				for(let i=1; i<=d; i++){
					tmp[i-1] = c[1][i]*i;
				}
				return tmp;
		}
	}
	return null;
}).join(" "));

